lib/routes/whu/rsgis.ts

```diff
@@ -0,0 +1,285 @@
+    const title = linkElement.text();
+    const href = linkElement.attr('href');
+    if (href === undefined) {
+        throw new Error('Cannot get link');
+    }
+    const external = checkExternal(href);
+    const link = external ? href : new URL(href, currentUrl).href;
+    const pubDate = element.find('div.date1').first().text();
+    return {
+        title,
+        link,
+        pubDate: timezone(parseDate(pubDate, 'YYYY-MM-DD'), +8),
+        description: title,
+        external,
+    };
+}
+
+async function getDetail(item: Post): Promise<DataItem | any> {
+    const link = item.link;
+    return link ? (await cache.tryGet(`whu:rsgis:${link}`, async () => {
+            if (item.external) {
+                item.description = `<a href="${link}">阅读原文</a>`;
+            } else {
+                const response = await ofetch(link);
+                const $ = load(response);
+                const title = $('div.content div.content_title h1').first().text();
+                const content = $('div.content div.v_news_content').first().html();
+                item.title = title;
+                item.description = content || '';
+            }
+            return item;
+        })) : item;
+}
+
+/**
+ * Process index type.
+ */
+async function handleIndex(): Promise<Array<Post>> {
+    const url = `${baseUrl}/index.htm`;
+    const response = await ofetch(url);
+    const $ = load(response);
+    // 学院新闻
+    const xyxwList: Array<Post> = $('div.main1 > div.newspaper:nth-child(1) > div.newspaper_list > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 通知公告
+    const tzggList: Array<Post> = $('div.main1 > div.newspaper:nth-child(2) > div.newspaper_list > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 学术动态
+    const xsdtList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(1) > ul.newspaper_list2 > li:nth-child(1) > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 学术进展
+    const xsjzList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(1) > ul.newspaper_list2 > li:nth-child(2) > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 教学动态
+    const jxdtList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(2) > div.newspaper_list2 > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 学工动态
+    const xgdtList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(3) > div.newspaper_list2 > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 组合所有新闻
+    const fullList = await Promise.all(
+        [...xyxwList, ...tzggList, ...xsdtList, ...xsjzList, ...jxdtList, ...xgdtList].map(async (item) => ({
+            ...item,
+            ...(await getDetail(item)),
```

Comment:
Since `item` passed into `getDetail` will be returned as well
https://github.com/DIYgod/RSSHub/blob/4b65c6e4323310df99601a4d8618bf7c3c200fb8/lib/routes/whu/rsgis.ts#L136-L137

You can returned `await getDetail(item)` directly without spreading `item` and `(await getDetail(item))`.